lib/Kconfig.debug

```diff
@@ -2607,12 +2607,18 @@ config RUST_OPT_LEVEL_0
 	  directly). Use the corresponding configuration options to control
 	  that instead, orthogonally.
 
+	  Note this level may cause excessive stack usage, which can lead to stack
+	  overflow and subsequent crashes.
```

Comment:
Perhaps it is best to add a mention to `CONFIG_THREAD_SHIFT` here (and for level 1), removing it from `Documentation/rust/arch-support.rst`, since other architectures may trigger the same problem sooner or later?

Reply:
I thought about that, but AFAIK powerpc is the only arch that supports changing the stack size, so I thought we didn't want to clutter this generic help with powerpc only details.

Reply:
Sounds good to me!